core/geometry/src/bspline/InterpolationCurve3d.ts

```diff
@@ -0,0 +1,191 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+/** @packageDocumentation
+ * @module Bspline
+ */
+
+import { Point3d, Vector3d } from "../geometry3d/Point3dVector3d";
+import { Geometry } from "../Geometry";
+import { Point3dArray } from "../geometry3d/PointHelpers";
+import { ProxyCurve } from "../curve/ProxyCurve";
+import { CurvePrimitive, CurvePrimitiveType } from "../curve/CurvePrimitive";
+import { BSplineCurve3d } from "./BSplineCurve";
+import { GeometryQuery } from "../curve/GeometryQuery";
+import { Transform } from "../geometry3d/Transform";
+import { GeometryHandler } from "../geometry3d/GeometryHandler";
+
+/**
+ * fitPoints and end condition data for [[InterpolationCurve3d]]
+ * @public
+ */
+export interface InterpolationCurve3dProps {
+   /** Order of the computed bspline.   (one more than degree) */
+  order?: number;
+  /** true if the bspline construction should be periodic */
+  closed?: boolean;
+  isChordLenKnots?: number;
+  isColinearTangents?: number;
+  isChordLenTangent?: number;
+  isNaturalTangents?: number;
+  /** optional start tangent.  Use of the tangent magnitude may be indicated by other flags. */
+  startTangent?: Vector3d;
+  /** optional end tangent.  Use of the tangent magnitude may be indicated by other flags. */
+  endTangent?: Vector3d;
+  /** Points that the curve must pass through */
+  fitPoints: Point3d[];
+  /** knots for curve fitting */
+  knots?: number[] | Float64Array;
+}
+/**
+ * Interpolating curve.
+ * * Derive from [[ProxyCurve]]
+ * * Use a [[BSplineCurve3d]] as the proxy
+ * *
+ * @public
+ */
+export class InterpolationCurve3d extends ProxyCurve  {
+  public curvePrimitiveType: CurvePrimitiveType;

```

Comment:
Is this intentionally mutable? When would someone set it to something other than "interpolationCurve"?

Reply:
I'll fix that.

Reply:
Cool, I'll resolve it once you push that fix.

Reply:
Should be `public readonly curvePrimitiveType = "interpolationCurve";`